server/enterprise/message_export/csv_export/csv_export.go

```diff
@@ -70,90 +72,120 @@ func CsvExport(rctx request.CTX, posts []*model.MessageExport, db shared.Message
 	})
 
 	if err != nil {
-		return warningCount, fmt.Errorf("unable to add header to the CSV export: %w", err)
+		return results, fmt.Errorf("unable to add header to the CSV export: %w", err)
 	}
 
+	// postAuthorsByChannel is a map so that we don't store duplicate authors
+	postAuthorsByChannel := make(map[string]map[string]shared.ChannelMember)
 	metadata := shared.Metadata{
-		Channels:         map[string]*shared.MetadataChannel{},
+		Channels:         p.ChannelMetadata,
 		MessagesCount:    0,
 		AttachmentsCount: 0,
-		StartTime:        0,
-		EndTime:          0,
+		StartTime:        p.BatchStartTime,
+		EndTime:          p.BatchEndTime,
 	}
 
-	postAuthorsByChannel := make(map[string]map[string]shared.ChannelMember)
+	channelsInThisBatch := make(map[string]bool)
+	for _, post := range p.Posts {
+		if post == nil {
+			results.IgnoredPosts++
+			rctx.Logger().Warn("ignored a nil post reference in the list")
+			continue
```

Comment:
Curious to understand when this would happen.

Reply:
No, it should never happen after the original `MessageExport` db call, and `preparePosts` which iterates it. Removed (and from the other paths (I'll get the one in GlobalRelay in that PR).